### PR TITLE
feat/Python: add CommitmentContextInput

### LINKED ISSUE #902: Add CommitmentInput - python
https://github.com/iotaledger/tips-draft/blob/tip45/tips/TIP-0045/tip-0045.md#commitment-input

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

Adds CommitmentContextInput to python

## Links to any relevant issues

Fixes #902

### CODE REVIEW

File: bindings/python/iota_sdk/types/context_input.py [hunk 1]
```diff
@@ -22,11 +23,31 @@ class ContextInput():
...
+
 @json
 @dataclass
 class BlockIssuanceCreditContextInput(ContextInput):
     """A Block Issuance Credit (BIC) Context Input provides the VM with context for the value of
     the BIC vector of a specific slot.
+
+    Attributes:
+        type: The type of context input.
+        account_id: The BIC of an account to use.
```
Comment (DaughterOfMars) [inline bindings/python/iota_sdk/types/context_input.py]: Huh?

Comment (kwek20) [reply bindings/python/iota_sdk/types/context_input.py]: huh :D

Comment (DaughterOfMars) [reply bindings/python/iota_sdk/types/context_input.py]: This comment doesn't make any sense 😆

Comment (thibault-martinez) [reply bindings/python/iota_sdk/types/context_input.py]: Yeah should just be ID

Comment (kwek20) [reply bindings/python/iota_sdk/types/context_input.py]: well its like this in the tip it seems.. hmm  

Made a comment, problem for another day.
https://github.com/iotaledger/tips-draft/pull/19/files#r1325846278

Comment (Thoralf-M) [inline bindings/python/iota_sdk/types/context_input.py]:
```suggestion
        account_id: The ID of the Account for which this input provides the BIC.
```

**Review by Thoralf-M — DISMISSED** [review]

File: bindings/python/iota_sdk/types/context_input.py [hunk 2]
```diff
@@ -22,11 +23,31 @@ class ContextInput():
...
 
 
+@json
+@dataclass
+class CommitmentContextInput(ContextInput):
+    """A Commitment Input allows referencing a commitment to a certain slot.
+
+    Attributes:
+        type: The type of context input.
+        commitment_id: The commitment identifier to reference to.
```
Comment (DaughterOfMars) [inline bindings/python/iota_sdk/types/context_input.py]: This comment makes sense but is poorly worded

**Review by Alex6323 — DISMISSED** [review]

File: bindings/python/iota_sdk/types/context_input.py (hunk 2 above)
Comment (thibault-martinez) [reply bindings/python/iota_sdk/types/context_input.py]: We addressing this?

Comment (kwek20) [reply bindings/python/iota_sdk/types/context_input.py]: What do you suggest to change to?

Comment (DaughterOfMars) [reply bindings/python/iota_sdk/types/context_input.py]: Probably just "The commitment identifier" is fine, right?

Comment (thibault-martinez) [inline bindings/python/iota_sdk/types/context_input.py]:
```suggestion
        commitment_id: The commitment identifier to reference.
```

File: bindings/python/iota_sdk/types/context_input.py [hunk 3]
```diff
@@ -22,11 +23,31 @@ class ContextInput():
     type: int
 
 
+@json
+@dataclass
+class CommitmentContextInput(ContextInput):
+    """A Commitment Input allows referencing a commitment to a certain slot.
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/types/context_input.py]:
```suggestion
    """A Commitment Context Input allows referencing a commitment to a certain slot.
```

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**